### ISSUE TITLE
Add optimizers with decoupled weight decay

### ISSUE THREAD

**Issue by tianxin1860** [PaddlePaddle/Paddle#14812]
[Fixing Weight Decay Regularization](https://arxiv.org/abs/1711.05101) 这篇论文指出了在使用Adam、SGDM等优化算法时，weight_decay与L2 regularizer并不是等价的，直接把Adam和L2Decay组合起来最终的效果会背离weight_decay的本意，希望尽快添加Adam、SGDM等优化算法的WeightDecay版本实现；

tensorflow实现参考: https://github.com/tensorflow/tensorflow/pull/17438

**Comment by lucywsq** [comment]
感谢您的建议，我们已经记录~为了保证更好的用户体验，我们将于今天内关闭此issue。若在关闭后您仍需跟进提问，可重新开启此问题，
我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持!

**Comment by lucywsq** [comment]
感谢您的建议，我们已经记录。
为了保证更好的用户体验，我们将于今天内关闭此issue。若在关闭后您仍需跟进提问，可重新开启此问题，
我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解感谢您对PaddlePaddle的支持!
